xtc/src/lib.rs

```diff
@@ -14,6 +14,53 @@ mod utils;
 #[cfg(test)]
 mod tests;
 
+// TODO(qti3e): Fix this mess when kit v0.5 is out. there shouldn't be any unsafe code.
+use crate::fee::compute_fee;
+use crate::ledger::Ledger;
+use ic_kit::ic;
+use ic_kit::macros::inspect_message;
+use ic_kit_sys::ic0;
+
+#[inspect_message]
+fn inspect_message() {
+    // get the raw message size
+    let message_size = unsafe { ic0::msg_arg_data_size() as usize };
+
+    let method_name = ic_cdk::api::call::method_name();
+
+    // exception: wallet_call should actually accept large payload, but only if user has
+    // the fee to pay.
+    if method_name == "wallet_call" {
```

Comment:
Should we do this only if message size is greater than our threshold? This way this block doesn't need to run every wallet call.  Very minor and I'm cool with leaving it as is.

Reply:
Addressed.